### PR TITLE
Standardize all context manager __exit__ methods

### PR DESCRIPTION
**Pull request by adamchainz**
# I have made things!

The Django-MySQL test suite uses a subclass of `CaptureQueriesContext`, which is how I found its `__exit__()` method was mistyped. This PR standardizes all `__exit__` methods to use the correct optional types, and fixes some minor issues on the way.

## Related issues

n/a

### CODE REVIEW

File: django-stubs/core/files/base.pyi [hunk 1]
```diff
@@ -1,5 +1,4 @@
-import os
```
Comment (adamchainz) [inline django-stubs/core/files/base.pyi:1]: unused import

File: django-stubs/test/testcases.pyi [hunk 2]
```diff
@@ -50,10 +51,15 @@ class _AssertTemplateUsedContext:
     context: ContextList = ...
     def __init__(self, test_case: Any, template_name: Any) -> None: ...
     def on_template_render(self, sender: Any, signal: Any, template: Any, context: Any, **kwargs: Any) -> None: ...
-    def test(self): ...
-    def message(self): ...
-    def __enter__(self): ...
-    def __exit__(self, exc_type: Any, exc_value: Any, traceback: Any): ...
+    def test(self) -> None: ...
```
Comment (adamchainz) [inline django-stubs/test/testcases.pyi:54]: returns `None`: https://github.com/django/django/blob/d62563cbb194c420f242bfced52b37d6638e67c6/django/test/testcases.py#L131-L137

File: django-stubs/test/testcases.pyi [hunk 3]
```diff
@@ -50,10 +51,15 @@ class _AssertTemplateUsedContext:
...
+    def message(self) -> str: ...
```
Comment (adamchainz) [inline django-stubs/test/testcases.pyi:55]: Removed in Django 4.1, but previously returned a `str`: https://github.com/django/django/commit/806efe912b846c1fde250c9321d8334b7517cd56#diff-7833da5b45a68d00834388d97dd5c4413e3796497c7bc5e0cc2621b08a2d0df1L111

File: django-stubs/utils/archive.pyi [hunk 4]
```diff
@@ -14,7 +14,7 @@ class Archive:
         exc_type: Optional[Type[BaseException]],
         exc_value: Optional[BaseException],
         traceback: Optional[TracebackType],
-    ) -> Optional[bool]: ...
```
Comment (adamchainz) [inline django-stubs/utils/archive.pyi:17]: never returned a `bool`: https://github.com/django/django/blob/d62563cbb194c420f242bfced52b37d6638e67c6/django/utils/archive.py#L88-L89

File: django-stubs/db/utils.pyi [hunk 5]
```diff
@@ -1,5 +1,5 @@
 from types import TracebackType
-from typing import Any, Dict, Iterable, Iterator, List, Optional, Type
```
Comment (adamchainz) [inline django-stubs/db/utils.pyi:2]: unused import

File: django-stubs/db/transaction.pyi [hunk 6]
```diff
@@ -1,4 +1,4 @@
-from contextlib import ContextDecorator, contextmanager
```
Comment (adamchainz) [inline django-stubs/db/transaction.pyi:1]: unused import

**Review by sobolevn — APPROVED** [review]
Thanks! Good stuff

**Pull request merged**